mlrun/db/httpdb.py

```diff
@@ -295,6 +295,15 @@ def connect(self, secrets=None):
                 config.valid_function_priority_class_names
                 or server_cfg.get("valid_function_priority_class_names")
             )
+            config.artifacts.calculate_hash = (
+                config.artifacts.calculate_hash
+                or server_cfg.get("calculate_artifact_hash")
+            )
```

Comment:
```suggestion
            config.artifacts.calculate_hash = (
                config.artifacts.calculate_hash if config.artifacts.calculate_hash is not None
                else server_cfg.get("calculate_artifact_hash")
            )
```

This is the correct implementation of the logic you mentioned previously. The way the code is currently written, if the client spec was changed to `False` it will always take the server-side config, so there's no way to override to `False`.